wren-ai-service/src/utils.py

```diff
@@ -101,8 +106,38 @@ def wrapper_timer(*args, **kwargs):
                     f.write("\n-----------------------\n")
                     f.write(test_records[-1])
 
-            return value
+            return result
 
         return func(*args, **kwargs)
 
     return wrapper_timer
+
+
+def async_timer(func):
+    load_env_vars()
+
+    async def process(func, *args, **kwargs):
+        if asyncio.iscoroutinefunction(func):
+            logger.info("this function is a coroutine: {}".format(func.__name__))
+            return await func(*args, **kwargs)
+        else:
+            logger.info("this is not a coroutine")
+            return func(*args, **kwargs)
+
+    @functools.wraps(func)
+    async def wrapper_timer(*args, **kwargs):
+        if os.getenv("ENABLE_TIMER", False):
+            startTime = time.perf_counter()
+            result = await process(func, *args, **kwargs)
+            endTime = time.perf_counter()
+            elapsed_time = endTime - startTime
+
+            logger.info(
+                f"{func.__qualname__} Elapsed time: {elapsed_time:0.4f} seconds"
+            )
+
+            return result
+
+        return await process(func, *args, **kwargs)
```

Comment:
the else-case in the process function will cause an error if it is not a coroutine, the python interpreter will throw the error like 
```
TypeError: object int can't be used in 'await' expression
```